source/common/http/filter_manager.cc

```diff
@@ -1064,6 +1064,19 @@ void FilterManager::encodeHeaders(ActiveStreamEncoderFilter* filter, ResponseHea
     }
   }
 
+  const auto status = HeaderUtility::checkRequiredResponseHeaders(headers);
```

Comment:
I need eyes from code owners if this is the right place to send local reply. I also considered [the top of ActiveStream::encodeHeaders](https://github.com/envoyproxy/envoy/blob/25d50667f96ee98d7f6cad8e190a3798ea5cd66f/source/common/http/conn_manager_impl.cc#L1352)

Reply:
Another suggestion would be to get the codec to return an error if the status header is missing 
https://github.com/envoyproxy/envoy/blob/25d50667f96ee98d7f6cad8e190a3798ea5cd66f/source/common/http/conn_manager_impl.cc#L1470

Reply:
Nice, that would be better!